lib/rubocop/cop/naming/predicate_name.rb

```diff
@@ -3,59 +3,107 @@
 module RuboCop
   module Cop
     module Naming
-      # Checks that predicate methods names end with a question mark and
+      # Checks that predicate method names end with a question mark and
       # do not start with a forbidden prefix.
       #
       # A method is determined to be a predicate method if its name starts
       # with one of the prefixes defined in the `NamePrefix` configuration.
       # You can change what prefixes are considered by changing this option.
       # Any method name that starts with one of these prefixes is required by
-      # the cop to end with a `?`. Other methods can be allowed by adding to
-      # the `AllowedMethods` configuration.
+      # the cop to end with a `?`.
       #
-      # NOTE: The `is_a?` method is allowed by default.
+      # Add method names to the `AllowedMethods` configuration list to allow
+      # them despite beginning with a forbidden prefix. By default, the `is_a?`
+      # method is allowed.
       #
       # If `ForbiddenPrefixes` is set, methods that start with the configured
-      # prefixes will not be allowed and will be removed by autocorrection.
+      # prefixes will not be allowed.
       #
       # In other words, if `ForbiddenPrefixes` is empty, a method named `is_foo`
-      # will register an offense only due to the lack of question mark (and will be
-      # autocorrected to `is_foo?`). If `ForbiddenPrefixes` contains `is_`,
-      # `is_foo` will register an offense both because the ? is missing and because of
-      # the `is_` prefix, and will be corrected to `foo?`.
+      # will register an offense only due to the lack of question mark (and should be
+      # corrected to `is_foo?`). If `ForbiddenPrefixes` contains `is_`, `is_foo` will
+      # register an offense both because the ? is missing and because of the `is_`
+      # prefix, and should be corrected to `foo?`.
       #
       # NOTE: `ForbiddenPrefixes` is only applied to prefixes in `NamePrefix`;
       # a prefix in the former but not the latter will not be considered by
       # this cop.
       #
-      # @example
+      # @example NamePrefix: ['is_', 'has_', 'have_'] (default)
       #   # bad
       #   def is_even(value)
       #   end
       #
-      #   def is_even?(value)
+      #   # good
+      #   def even?(value)
```

Comment:
Btw, without matching "ForbiddenPrefixes" `is_even?` will be good as well, which I think we should mention for clarity.

Reply:
Yeah, it's a little tricky to demonstrate the combination of configurations. I pushed up an update which I hope makes it more clear.

While doing that, I also thought it would be nice to add a config validation to automatically check the config rather than trust that people actually read the docs. It seems possible to really streamline the prose at the top and lean on the examples and the config validation to demonstrate the nuances.

Feedback welcome of course.